lib/screenplay/screens_config.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Screenplay.ScreensConfig do
+  @moduledoc """
+  Supervisor and public interface for fetching and updating Screens data.
+  """
+
+  use Supervisor
+
+  alias Screenplay.ScreensConfig.Cache
+  alias Screenplay.ScreensConfig.Fetch.Fetcher
+  alias ScreensConfig.Screen
+
+  @spec start_link(any()) :: Supervisor.on_start()
+  def start_link(_) do
+    Supervisor.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @impl true
+  def init(_) do
+    children = [
+      Cache,
+      Fetcher
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  @spec screen(String.t()) :: Screen.t() | nil
+  def screen(screen_id) do
+    Cache.get(screen_id)
+  end
+
+  @spec screen_ids() :: list(String.t())
+  def screen_ids do
+    Cache.all(nil, return: :key)
+  end
+
+  def screens do
+    Cache.all(nil, return: {:key, :value})
+  end
+
+  @spec update_cache(list({String.t(), Screen.t()})) :: :ok
+  def update_cache(screens) do
+    Cache.put_all(screens)
```

Comment:
If I'm understanding the docs correctly, this would _not_ remove screens whose IDs are no longer present in the input, which is a perhaps undesirable change in behavior from the original version.

Reply:
Ahh yes thank you. I should throw a `delete_all` in there first.

Reply:
I don't think that would be a good way of handling this, since it would result in the store being empty for a brief interval — if another process accessed it during that time, it would see no screens in the config. This same pattern was behind some bugs we used to have (/maybe still have) in the V3 API where requests would very occasionally return `[]` instead of the correct list of resources. A better approach would be to determine the set of keys that are present in the existing data but not in the new data, and delete only those.

Reply:
Very good point. Can definitely make that happen.